src/components/new-request/submit-request/reserve-submit.vue

```diff
@@ -64,7 +72,25 @@ export default class ReserveSubmitButton extends Vue {
     this.cancelAnalyzeName('NamesCapture')
   }
 
-  handleSubmit () {
+  getData (type: string): any {
+    if (this.getAssumedName) type = 'assumed'
+    let data: any
+    switch (type) {
+      case 'assumed':
+      case 'draft':
+        data = this.getDraftNameReservation
+        break
+      case 'conditional':
+        data = this.getConditionalNameReservation
+        break
+      case 'reserved':
+        data = this.getReservedNameReservation
```

Comment:
Are these asynchronous?  ie could `data` be returned prior to the request resolving?

Reply:
I will check. This logic was taken from somewhere else I just repurposed it

Reply:
they are getters for store data so they shouldn't be but I will double check

Reply:
Lots of code for these but everything is just getting data from the store / manipulating it. No api calls or anything

Reply:
kk sounds good! Thanks for checking 👍